### ISSUE TITLE
Add support for scala 2.11

### ISSUE THREAD

**Issue by pdharma** [softwaremill/tapir#59]
I tried to use tapir with scala 2.11 but cannot locate any public jar files for that version of Scala. Would you be able to support this?

**Comment by adamw** [comment]
I didn't check, should be possible. Maybe you can try creating a cross-build and checking if there are any errors?

**Comment by ghostbuster91** [comment]
I did some work on that here: 22690f8, still having one compilation error : 
```
[error] /home/kghost/workspace/tapir/server/http4s-server/src/main/scala/tapir/server/http4s/EndpointToHttp4sServer.scala:79:35: polymorphic expression cannot be instantiated to expected type;
[error]  found   : [G[_], A]G[A]
[error]  required: org.http4s.HttpRoutes[F]
[error]     (which expands to)  cats.data.Kleisli[[β$0$]cats.data.OptionT[F,β$0$],org.http4s.Request[F],org.http4s.Response[F]]
[error]       case Some(routes) => routes.reduceK
[error]                                   ^
[error] one error found
```
I tried writing it by hand but then it complains about not being able to find implicit instance for semigroupK:
```
G: cats.SemigroupK[[γ$1$]cats.data.Kleisli[[β$0$]cats.data.OptionT[F,β$0$],org.http4s.Request[F],γ$1$]]
[error]         new NestedReducibleOps[NonEmptyList, Kleisli[OptionT[F, ?], Request[F], ?], Response[F]](routes).reduceK
[error]                                                                                                          ^
[error] one error found

```
And I don't have any idea how to fix it.

**Comment by adamw** [comment]
@ghostbuster91 no idea, I would try providing the arguments to `reduceK` explicitly. Starting from `reduceK(implicitly, implicitly)`, and then replacing the argument that fails to be found with a specific instance (as suggested e.g. by intellij).

OR ask on http4s gitter?

**Comment by adamw** [comment]
Released in 0.7.8
